crypto/BRCryptoWalletManager.c

```diff
@@ -66,18 +86,43 @@ static void
 cwmGetBlockNumberAsBTC (BRWalletManagerClientContext context,
                         BRWalletManager manager,
                         int rid) {
-    BRCryptoWalletManager cwm = context;
-    cwm->client.btc.funcGetBlockNumber (cwm->client.context, manager, rid);
+    BRCryptoWalletManager cwm = cryptoWalletManagerTake (context);
```

Comment:
General pattern is that before invoking the callback, we create a callback record that tracks any state needed in the associated "announce" function to perform its task.